src/main/java/hudson/plugins/jira/JiraSite.java

```diff
@@ -909,13 +920,12 @@ public boolean existsIssue(String id) {
      * @return A set of JiraVersions
      */
     public Set<ExtendedVersion> getVersions(String projectKey) {
-        JiraSession session = getSession();
-        if (session == null) {
+        if (this.jiraSession == null) {
             LOGGER.warning("Jira session could not be established");
             return Collections.emptySet();
         }
 
-        return new HashSet<>(session.getVersions(projectKey));
+        return new HashSet<>(this.jiraSession.getVersions(projectKey));
```

Comment:
Why `this.` is added everywhere? there is no other session here, so that shouldn't be needed

Reply:
ah yes good point. There is maybe a problem here. I'm based on the fact than all consumer use the static `JiraSite.get()` so the session is created there but this can we wrong we can have some access via Builder or constructor. I need to change that....

Reply:
oh I'm lost you mean the keyword `this`? I don't see any problem using it

Reply:
There are 2 things here:

1. using `this.jiraSession` instead of just `jiraSession` - I'm not against it, but firstly, there is no need to use it here and secondly that's inconsistent - you're also touching `projects` fields but it's not changed to `this.projects`.

2. Maybe I'm missing something, but you removed line 912 - so in case `this.jiraSession` is not yet created, this would fail - or was it moved somewhere else and I just don't see it?

Reply:
please read new first line of the class level javadoc :)

Reply:
LoL, dunno why, but GH somehow hid that change from me :+1: 
But this comment is about getting JiraSite and here (in line 912) we are talking about getting *session*, that will be null when JiraSession is instantiated

Reply:
please look at get methods of JiraSite https://github.com/jenkinsci/jira-plugin/pull/249/files#diff-974b061f6eda1c5088eaa93c765904edR1364

Reply:
ah, that's well hidden in the code - so that's actually missing from the javadoc for those two methods as they should say "and creates automatically jiraSession for each jiraSite found".